sticker/src/tensorflow/dataset.rs

```diff
@@ -151,3 +146,65 @@ where
         Some(Ok(builder))
     }
 }
+
+/// Trait providing adapters for `conllx::io::Sentences`.
+pub trait SentenceIter: Sized {
+    fn filter_by_len(self, max_len: usize) -> LengthFilter<Self>;
+}
+
+impl<I> SentenceIter for I
+where
+    I: Iterator<Item = Result<Sentence, Error>>,
+{
+    fn filter_by_len(self, max_len: usize) -> LengthFilter<Self> {
+        LengthFilter {
+            inner: self,
+            max_len,
+        }
+    }
+}
+
+/// An Iterator adapter filtering sentences by maximum length.
+pub struct LengthFilter<I> {
+    inner: I,
+    max_len: usize,
+}
+
+impl<I> Iterator for LengthFilter<I>
+where
+    I: Iterator<Item = Result<Sentence, Error>>,
+{
+    type Item = Result<Sentence, Error>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        while let Some(sent) = self.inner.next() {
+            let len = sent.as_ref().map(|s| s.len()).unwrap_or(0);
```

Comment:
`sent` is a `Result<Sentence, Error>`, this will result in `len == 0` when the `Result` is `Err`. This is fine here, since the iterator will properly return `Some(Err(_))` below. But it is probably good to add a small comment above this line pointing that out.

Reply:
Added a comment